stablehlo/dialect/TypeInference.cpp

```diff
@@ -946,9 +958,16 @@ LogicalResult inferSliceOp(Optional<Location> location, Value operand,
   SmallVector<int64_t, 4> limit(limitIndices.getValues<int64_t>());
   SmallVector<int64_t, 4> strideVals(strides.getValues<int64_t>());
 
+  llvm::SmallVector<int64_t> inputBounds;
+  BoundedDialectInterface* dialect = getBounds(rankedTy, inputBounds);
+
   SmallVector<int64_t, 4> shape;
   shape.reserve(rank);
+  SmallVector<int64_t> resultBounds;
   for (int64_t i = 0, e = rank; i != e; i++) {
+    if (!inputBounds.empty())
+      resultBounds.push_back(
+          inferSliceDim(inputBounds[i], start[i], limit[i], strideVals[i]));
```

Comment:
While we're here, let's fix `inferSliceDim` to work with `ShapedType::kDynamicSize` instead of `-1`. It is likely that `-1` will stop working in a few weeks.

Reply:
Fixed. I do not touch any "-1" outside `inferSliceDim`, as I assume there is another issue tracking that already.

Reply:
There is: #327.